### PR TITLE
add new plugin evict-pod

### PR DESCRIPTION
**Pull request by rajatjindal**
Hi Krew Team,

I would like to add a new plugin evict-pod. This plugin is useful when testing pod disruption budgets.

Thanks
Rajat Jindal

### CODE REVIEW

File: plugins/evict-pod.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        arch: amd64
+    uri: https://github.com/rajatjindal/kubectl-evict-pod/releases/download/v0.0.1/linux-amd64-v0.0.1.tar.gz
+    sha256: 53f4cf2161921fce47e29b2db660088f075d2337dbbb790ec19be93fc475b025
+    files:
+    - from: "*"
+      to: "."
+    bin: kubectl-evict-pod
+  shortDescription: Evicts the given pod.
+  description: |
+    Usage:
```
Comment (ahmetb) [inline plugins/evict-pod.yaml]: This usage string isnt useful. Consider removing.

File: plugins/evict-pod.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      matchLabels:
+        os: linux
+        arch: amd64
+    uri: https://github.com/rajatjindal/kubectl-evict-pod/releases/download/v0.0.1/linux-amd64-v0.0.1.tar.gz
+    sha256: 53f4cf2161921fce47e29b2db660088f075d2337dbbb790ec19be93fc475b025
+    files:
+    - from: "*"
+      to: "."
+    bin: kubectl-evict-pod
+  shortDescription: Evicts the given pod.
```
Comment (ahmetb) [inline plugins/evict-pod.yaml]: Don’t need trailing period

File: plugins/evict-pod.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    - from: "*"
+      to: "."
+    bin: kubectl-evict-pod
+  - selector:
+      matchLabels:
+        os: linux
+        arch: amd64
+    uri: https://github.com/rajatjindal/kubectl-evict-pod/releases/download/v0.0.1/linux-amd64-v0.0.1.tar.gz
+    sha256: 53f4cf2161921fce47e29b2db660088f075d2337dbbb790ec19be93fc475b025
+    files:
```
Comment (ahmetb) [inline plugins/evict-pod.yaml]: Fyi i think 

From:* to:.

Is the default now. You don’t need to list explicitly.

**Comment by ahmetb** [comment]
Initially, I didn't believe kubectl didn't have a command (or a `delete` flag) for this. But it seems to be the case that docs use the REST API to achieve this: https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#the-eviction-api

Maybe also grab a slot in SIG CLI bi-weekly meetings and present this plugin (why you built it, would you want to make it part of kubectl etc).

**Comment by ahmetb** [comment]
/hold

this plugin also doesn't import auth plugins

```
$ kubectl evict-pod
Error: No Auth Provider found for name "gcp"
```

**Comment by rajatjindal** [comment]
Hi Ahmet,

I've published a new release with the import of auth. if you get a chance, can you verify if this fixes the error?

https://github.com/rajatjindal/kubectl-evict-pod/releases/tag/v0.0.2-prerelease

or, I will setup my own cluster later next week as I will be traveling for most of this week. I hope its ok to have open PR until I fix this issue (latest by sometime next week).

Thanks
Rajat Jindal

**Comment by rajatjindal** [comment]
replaced by https://github.com/kubernetes-sigs/krew-index/pull/293

**Pull request closed without merging**